### ISSUE TITLE
bug: vercel ai sdk integration does not reliably pick up `functionId` as the generation/span name

### ISSUE THREAD

**Issue by marcklingen** [langfuse/langfuse#3140]
### Describe the bug

In a previous version, the functionId was recorded as the trace name (as per documentation). Since initial release, this behavior changed. Langfuse should pick up the the functionId as the generation name to allow for filtering on the generations table and dashboards.

### To reproduce

Follow this guide: https://langfuse.com/docs/integrations/vercel-ai-sdk#pass-custom-attributes

### SDK and container versions

"ai": "^3.3.20"

### Are you interested to contribute a fix for this bug?

Yes

**Comment by Averylamp** [comment]
Hi, I was looking at how to set a span's name with the Vercel AI SDK and found this related issue.  From my understanding, functionId will always replace the trace's name but not the span's name in the tracing?
I couldn't figure out how I would be able to set a span's name with LangFuse as it seems to always override the entire trace's name.  If this is also not the correct place to comment I am happy to open another issue.  Thanks

**Comment by marcklingen** [comment]
great question! @hassiebp is the span name configurable at the moment? otherwise this would be a good idea post to keep track of this improvement to this integration

**Comment by david8z** [comment]
Looks for the same as @Averylamp here

**Comment by david8z** [comment]
Created the RP to support it:
https://github.com/langfuse/langfuse-js/pull/450

cc/ @marcklingen @hassiebp

**Comment by hassiebp** [comment]
@Averylamp @david8z This is currently not possible. The AI SDK is in full control of all the spans + nesting below the trace level, and the approach in https://github.com/langfuse/langfuse-js/pull/450 would override _all_ span names with the provided name at the top level. This change should 'live' at the level of AI SDK telemetry, exposing better control over generated spans and span names. 

Could you please explain what you aim to achieve with the custom span names?

**Comment by Averylamp** [comment]
Yeah for sure. 
So custom span names would be useful because we have a multi stage pipeline that runs in parallel so simply it’s hard to tell which stage is which by looking at the autogenerated span names which all say the base command ai.streamText.  You end up having to just click on everything to find the right one that you want. 
 I think also it’d be useful because the generations tab is also now useless because everything says ai.streamText because nothing is nameable without changing the entire trace name. 
Because things run in parallel, naming things also makes it inconsistent in what the final name ends up being because they all override each other.  
I imagine the right level for this to live is just at the extra metadata that ai sdk allows to pass that simply overrides just the span name but not the entire trace name.

**Comment by david8z** [comment]
@Averylamp covered it perfectly. Just to ilustrate it I see this and I have no idea what I am seeing they all have the exact same name:

![image](https://github.com/user-attachments/assets/0e0f29fb-ad92-4743-8f0c-2b8d63ab0540)

In my ideal words
![image](https://github.com/user-attachments/assets/b9991e89-c4e0-4e9d-a0ac-834832e4a0a4)

**Comment by david8z** [comment]
Any update with this?

Why does my PR brake anything i could iterate it if needed. cc/ @hassiebp

**Comment by hassiebp** [comment]
Thanks for providing the detail! Prior to moving forward with the PR, could you please share the example code that created those parallel runs / traces? That would help to understand whether this issue is best addressed within Langfuse or the AI SDK.

Fundamentally, span names are managed by the AI SDK, as they are 'in charge' of the child spans and their nesting. So an override at the trace level prior to the execution cannot name a specific span as we have no prior knowledge on the resulting span tree. Overrides can be only be easily applied for the root span, or more fiddly by some conditional logic.

**Comment by david8z** [comment]
Usnure I am following you here @hassiebp could you rephrase it.

This is generated basically by running multiple generations with the the aiSdk all providing the same traceId on the metadata.


So:
```
await generateObject({
   ...
   experimental_telemetry: {
     isEnabled: true,
     metadata: {
      langfuseTraceId: 1
     }
  }
})

await generateObject({
   ...
   experimental_telemetry: {
     isEnabled: true,
     metadata: {
      langfuseTraceId: 1
     }
  }
})
```

This will push two generations to the same trace with the identical span name so not identifiable until you go in.

**Comment by david8z** [comment]
Any updates on this? cc/ @hassiebp

**Comment by hassiebp** [comment]
Yup PR is open, soon to be merged and released :) Thanks for following up @david8z !

**Comment by hassiebp** [comment]
@david8z @Averylamp We have just released support to group multiple Vercel AI SDK executions under one trace. Please upgrade `langfuse-vercel` [to latest](https://github.com/langfuse/langfuse-js/releases/tag/v3.30.4) and check out [our docs here for more detail](https://langfuse.com/docs/integrations/vercel-ai-sdk#group-multiple-executions-in-one-trace)
